src/dbnode/persist/fs/seek_manager.go

```diff
@@ -136,16 +156,33 @@ type seekersByTime struct {
 	sync.RWMutex
 	shard    uint32
 	accessed bool
-	seekers  map[xtime.UnixNano]rotatableSeekers
+	seekers  map[xtime.UnixNano]*rotatableSeekers
 }
 
 // rotatableSeekers is a wrapper around seekersAndBloom that allows for rotating
 // out stale seekers. This is required so that the active seekers can be rotated
 // to inactive while the seeker manager waits for any outstanding stale seekers
 // to be returned.
 type rotatableSeekers struct {
-	active   seekersAndBloom
-	inactive seekersAndBloom
+	// Active seekers for the latest volume which will serve queries.
+	active seekersAndBloom
+	// We hold on to inactive seekers while we wait for any borrowed seekers to be returned.
+	// These are cleaned up after all borrowed seekers have been returned.
+	// Inactive seekers are created from active seekers under two conditions:
+	//   - When a new volume is flushed to disk.
+	//   - When a previously owned shard is given away during a node add.
+	inactive []seekersAndBloom
+}
+
+func (r *rotatableSeekers) removeInactiveAtIndex(idx int) {
```

Comment:
Worth doing an index check and returning error if out of bounds?

Reply:
Ended up removing this function.